lms/static/js/spec/search/search_spec.js

```diff
@@ -611,19 +611,7 @@ define([
                 $('.search-button').trigger('click');
                 // cancel search
                 $('.cancel-button').trigger('click');
-                AjaxHelpers.respondWithJson(requests, {
-                    total: 1337,
-                    access_denied_count: 12,
-                    results: [{
-                        data: {
-                            location: ['section', 'subsection', 'unit'],
-                            url: '/some/url/to/content',
-                            content_type: 'text',
-                            excerpt: 'this is a short excerpt',
-                            course_name: ''
-                        }
-                    }]
-                });
+                AjaxHelpers.skipResetRequest(requests);
```

Comment:
How does this test work? I'm guessing that this skipResetRequest will cancel the ajax request triggered by clicking the search button, but let the request triggered by the cancel button click go through? 
Could you please add a comment to the code, it seems a bit tricky to understand.